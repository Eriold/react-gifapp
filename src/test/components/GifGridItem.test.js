import { shallow } from 'enzyme'
import React from 'react'
import '@testing-library/jest-dom'
import { GifGridItem } from '../../Components/GifGridItem'

describe('render <GifGridItem />', () => {

  const title = 'A title'
  const url = 'https://localhost/algo.jpg'

  const wrapper = shallow(
    <GifGridItem
      title={title}
      url={url}
    />)

  test('show <GifGridItem /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('evaluate <p> with title', () => {
    const showTitleText = wrapper.find('p').text().trim();
    expect(showTitleText).toBe(title)
  })

  test('evaluate <img> with url and title', () => {
    const img = wrapper.find('img');
    // console.log(img.props('src'))
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('cant to have animate_fadeIn in className ', () => {
    const div = wrapper.find('div');
    // console.log(div.prop('className'));
    const classNameDiv = div.prop('className')
    expect(classNameDiv.includes('animate__bounce')).toBe(true)
  })

})