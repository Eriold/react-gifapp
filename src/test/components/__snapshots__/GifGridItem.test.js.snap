// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <GifGridItem /> show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="A title"
    src="https://localhost/algo.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
